agent/api/task/service_connect_test.go

```diff
@@ -0,0 +1,261 @@
+		} else {
+			ingressConfig = fmt.Sprintf(`\"listenerPort\":%s`, testBridgePortDefault)
+		}
+	}
+
+	testAWSVPCServiceConnectConfig := fmt.Sprintf(`"{\"statsConfig\":{\"path\":\"%s\"},\"egressConfig\":{\"listenerName\":\"%s\",\"vip\":{\"ipv4Cidr\":\"%s\",\"ipv6Cidr\":\"%s\"}},\"dnsConfig\":[{\"hostname\":\"%s\",\"address\":\"%s\"}],\"ingressConfig\":[{%s}]}"`, testStatsConfigPath, testOutboundListenerName, testIPV4Cidr, testIPV6Cidr, testHostName, testAddress, ingressConfig)
+
+	return &ecsacs.Task{
+		Attachments: []*ecsacs.Attachment{
+			{
+				AttachmentArn: strptr("attachmentArn"),
+				AttachmentProperties: []*ecsacs.AttachmentProperty{
+					{
+						Name:  strptr(serviceConnectAttProSCContainerName),
+						Value: strptr(testServiceConnectContainerName),
+					},
+					{
+						Name:  strptr(serviceConnectAttProSCConfigName),
+						Value: strptr(testAWSVPCServiceConnectConfig),
+					},
+				},
+				AttachmentType: strptr(serviceConnectAttType),
+			},
+		},
+	}
+}
+
+func getTestServiceConnectConfig(scHostName, scAddress, scOutListenerName, scIPV4Cidr, scIPV6Cidr, scPath string, scIngressConfig []IngressConfigEntry) *ServiceConnectConfig {
+	return &ServiceConnectConfig{
+		ContainerName: testServiceConnectContainerName,
+		StatsConfig: StatsConfig{
+			Path: scPath,
+		},
+		IngressConfig: scIngressConfig,
+		EgressConfig: EgressConfig{
+			ListenerName: scOutListenerName,
+			VIP: VIP{
+				IPV4CIDR: scIPV4Cidr,
+				IPV6CIDR: scIPV6Cidr,
+			},
+		},
+		DNSConfig: []DNSConfigEntry{
+			{
+				HostName: scHostName,
+				Address:  scAddress,
+			},
+		},
+	}
+}
+
+func TestExtractSCConfigFromACSAttachment(t *testing.T) {
+	tt := []struct {
+		testName      string
+		ingressConfig []IngressConfigEntry
+		testACSTask   *ecsacs.Task
+	}{
+		{
+			testName: "AWSVPC default case",
+			ingressConfig: []IngressConfigEntry{
+				{
+					InterceptPort: aws.Uint16(8080),
+					ListenerName:  testInboundListenerName,
+				},
+			},
+			testACSTask: getTestACSTaskAttachment(awsvpcMode, false),
+		},
+		{
+			testName: "AWSVPC port override case",
+			ingressConfig: []IngressConfigEntry{
+				{
+					ListenerPort: uint16(8080),
+				},
+			},
+			testACSTask: getTestACSTaskAttachment(awsvpcMode, true),
+		},
+		{
+			testName: "Bridge default case",
+			ingressConfig: []IngressConfigEntry{
+				{
+					ListenerPort: uint16(15000),
+				},
+			},
+			testACSTask: getTestACSTaskAttachment(bridgeMode, false),
+		},
+		{
+			testName: "Bridge port override case",
+			ingressConfig: []IngressConfigEntry{
+				{
+					ListenerPort: uint16(8080),
+					HostPort:     aws.Uint16(8080),
+				},
+			},
+			testACSTask: getTestACSTaskAttachment(bridgeMode, true),
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.testName, func(t *testing.T) {
+			testSCConfig := getTestServiceConnectConfig(testHostName, testAddress, testOutboundListenerName, testIPV4Cidr, testIPV6Cidr, testStatsConfigPath, tc.ingressConfig)
```

Comment:
another nit: let's change this to `expectedTestSCConfig`

Reply:
testSCConfig is renamed to expectedTestSCConfig in commit 8e69368bc5c1d7bd3ab71391103f4b6a69f06054. Thanks!